### PR TITLE
Fix pdh-based go checks, when variable instances are added & removed

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Previously, available instances were only computed at `config` time; so if an instance was added (e.g. a drive plugged in), then it was missed by subsequent checks

### Motivation

customer issue

### Additional Notes

Also adds mocked version of PDH functions, and adds some unit tests.

### CODE REVIEW

File: pkg/logs/pipeline/registry.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"Version":2,"Registry":{}}
```
Comment (achntrl) [inline pkg/logs/pipeline/registry.json]: Not sure you wanted to commit this file

**Review by hush-hush — COMMENTED** [review]
Can't really review the PDH logic, left a few comments

File: pkg/util/winutil/pdhutil/pdhcounter.go [hunk 2]
```diff
@@ -28,60 +31,35 @@ type CounterInstanceVerify func(string) bool
...
+	verifyfn             CounterInstanceVerify
 }
 
-// GetCounterSet returns an initialized PDH counter set.
-func GetCounterSet(className string, counterName string, instanceName string, verifyfn CounterInstanceVerify) (*PdhCounterSet, error) {
-	var p PdhCounterSet
-	p.countermap = make(map[string]PDH_HCOUNTER)
-	var err error
+// Initialize initializes a counter set object
+func (p *PdhCounterSet) Initialize(className string) (err error) {
```
[the remark this answers is not in the document]
Comment (derekwbrown) [reply pkg/util/winutil/pdhutil/pdhcounter.go]: If I understand right, only `ndxlist` would be initialized on line 60... but I can change it...

File: pkg/util/winutil/pdhutil/pdhcounter.go [hunk 3]
```diff
@@ -91,71 +69,145 @@ func GetCounterSet(className string, counterName string, instanceName string, ve
...
-			}
-			path, err := p.MakeCounterPath("", counterName, inst, allcounters)
-			if err != nil {
-				continue
-			}
-			var hc PDH_HCOUNTER
-			winerror = PdhAddCounter(p.query, path, uintptr(0), &hc)
-			if ERROR_SUCCESS != winerror {
+	winerror := pfnPdhAddCounter(p.query, path, uintptr(0), &p.singleCounter)
+	if ERROR_SUCCESS != winerror {
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply pkg/util/winutil/pdhutil/pdhcounter.go]: OK !

**Review by hush-hush — APPROVED** [review]

**Pull request merged**